lib/segment/src/index/hnsw_index/point_scorer.rs

```diff
@@ -36,8 +37,12 @@ use crate::vector_storage::{
 ///                                                              - Scoring logic
 ///                                                              - Complex queries
 /// ```
-pub struct FilteredScorer<'a> {
-    raw_scorer: Box<dyn RawScorer + 'a>,
+pub type FilteredScorer<'a> = FilteredScorerImpl<'a, Box<dyn RawScorer + 'a>>;
+
+pub type FilteredQuantizedScorer<'a> = FilteredScorerImpl<'a, Box<dyn QueryScorerBytes + 'a>>;
```

Comment:
Can we redesign RawScorer to avoid this definition? We did a great job of keeping all our abstractions in the strong abstraction layer and breaking them here. We don't have here a separate definition for multivectors, for Discovery/Context search, quantizations, etc. Even if the price of such abstraction is a bunch of functions with panics, it still keeps all abstraction layers together and allows here to forget about the kinds of scorers

Reply:
I find this topic critical enough to place merge blocker

Reply:
@generall need your help here. I guess we want to have support for new scoring in the `FilteredScorer` for exact search/low cardinality search etc, and won't remove this even after finishing the milestone.

Reply:
I don't have a strong opinion, please resolve among yourself.

Reply:
After the discussion, we decided to keep it as is because there is no better idea